src/Swarm/Game/State.hs

```diff
@@ -361,6 +365,10 @@ robotsInArea o d gs = map (rm IM.!) rids
   rl = gs ^. robotsByLocation
   rids = concatMap IS.elems $ getElemsInArea o d rl
 
+-- | The base robot, if it exists.
+baseRobot :: Traversal' GameState Robot
+baseRobot = robotMap . ix 0
```

Comment:
Eventually I want to stop assuming that the base is always ID 0, and this refactoring is a great first step!